src/DenoHTTPWorker.ts

```diff
@@ -0,0 +1,524 @@
+      for (let [key] of Object.entries(this._options.denoUnstable).filter(
+        ([_key, val]) => val
+      )) {
+        runArgs.push(
+          `--unstable-${key.replace(/[A-Z]/g, (m) => "-" + m.toLowerCase())}`
+        );
+      }
+    }
+    addOption(runArgs, "--cached-only", this._options.denoCachedOnly);
+    addOption(runArgs, "--no-check", this._options.denoNoCheck);
+    addOption(
+      runArgs,
+      "--unsafely-ignore-certificate-errors",
+      this._options.unsafelyIgnoreCertificateErrors
+    );
+    if (this._options.location) {
+      addOption(runArgs, "--location", [this._options.location]);
+    }
+
+    if (this._options.denoV8Flags.length > 0) {
+      addOption(runArgs, "--v8-flags", this._options.denoV8Flags);
+    }
+
+    if (this._options.denoImportMapPath) {
+      addOption(runArgs, "--import-map", [this._options.denoImportMapPath]);
+    }
+
+    if (this._options.denoLockFilePath) {
+      addOption(runArgs, "--lock", [this._options.denoLockFilePath]);
+    }
+
+    if (this._options.permissions) {
+      addOption(runArgs, "--allow-all", this._options.permissions.allowAll);
+      if (!this._options.permissions.allowAll) {
+        addOption(
+          runArgs,
+          "--allow-net",
+          typeof this._options.permissions.allowNet === "boolean"
+            ? this._options.permissions.allowNet
+            : this._options.permissions.allowNet
+            ? [...this._options.permissions.allowNet, allowAddress]
+            : [allowAddress]
+        );
+        // Ensures the `allowAddress` isn't denied
+        const deniedAddresses = this._options.permissions.denyNet?.filter(
+          (address) => address !== allowAddress
+        );
+        addOption(
+          runArgs,
+          "--deny-net",
+          // Ensures an empty array isn't used
+          deniedAddresses?.length ? deniedAddresses : false
+        );
+        addOption(runArgs, "--allow-read", this._options.permissions.allowRead);
+        addOption(
+          runArgs,
+          "--allow-write",
+          this._options.permissions.allowWrite
+        );
+        addOption(runArgs, "--allow-env", this._options.permissions.allowEnv);
+        addOption(
+          runArgs,
+          "--allow-plugin",
+          this._options.permissions.allowPlugin
+        );
+        addOption(
+          runArgs,
+          "--allow-hrtime",
+          this._options.permissions.allowHrtime
+        );
+      }
+    }
+
+    this._process = spawn(
+      this._options.denoExecutable,
+      ["run", ...runArgs, this._options.denoBootstrapScriptPath, ...scriptArgs],
+      this._options.spawnOptions
+    );
+    this._process.on("exit", (code: number, signal: string) => {
+      console.log("process exited");
+    });
+
+    this._stdout = <Readable>this._process.stdout;
+    this._stderr = <Readable>this._process.stderr;
+
+    if (this._options.logStdout) {
+      this._stdout.setEncoding("utf-8");
+      this._stdout.on("data", (data) => {
```

Comment:
So, I think, this is going to be wacky, and it depends on implementation, but `data` is not guaranteed to be a line - it can be multiple lines. I _think_ that if you get `deno-vm-port 100\nfoo` this'll still succeed because of JavaScript's nutty implementation of parseInt, but… if the next line starts with a number, it'll cause some chaos.